### PR TITLE
Week 2 Step 4 Post

### PR DESCRIPTION
**Pull request by maggiekoesno**

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -32,11 +32,106 @@ function addRandomFact() {
...
+    var commenter;
+
+    for (i = 0; i < comments.length; i++) {
+        commentDiv = document.createElement("div");
+        commentDiv.className = "comment-container";
+
+        paragraph = document.createElement("p");
+        paragraph.className = "commenter";
+        commenter = comments[i].commenter;
+        if (commenter === ""){
```
Comment (dhritiwasan14) [inline portfolio/src/main/webapp/script.js:86]: You could try commenter.isEmpty() here

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -32,11 +32,106 @@ function addRandomFact() {
...
+        paragraph.className = "comment-message";
+        node = document.createTextNode(comments[i].commentMessage);
+        paragraph.appendChild(node);
+        commentDiv.appendChild(paragraph);
+
+        replyForm = document.createElement("form");
+        replyForm.action = "/comments";
+        replyForm.method = "POST";
+
+        replyForm.appendChild(createInputElement("commenter", "Name (Optional)", "20", "20", false));
```
Comment (dhritiwasan14) [inline portfolio/src/main/webapp/script.js:103]: perhaps you can abstract this logic to createAddNewCommentElement()?

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -29,6 +29,16 @@ <h1> Hello!</h1>
                 <button onclick="addRandomFact()"> Tell me more </button>
                 <div id="fact-container"></div>
             </div>
+            <div class="subcontent" id="comments">
+                <p>Leave me a comment!</p>
+                <form name="2" action="/comments" method="POST">
+                    <input type="text" name="commenter" placeholder="Name (Optional)" size = "20" maxlength="20">
+                    <input type="text" name="comment-message" placeholder="Comment" size = "100" maxlength="400" required> 
+                    <input type="hidden" name="parent-comment" value="0"> 
+                    <input type="submit" value="Comment">
```
Comment (dhritiwasan14) [inline portfolio/src/main/webapp/index.html:38]: Oh did you know that having an input form lets you post by just clicking enter on the keyboard without clicking any submit button? This also works btw! :)

Comment (maggiekoesno) [reply portfolio/src/main/webapp/index.html:38]: I did not know that, good to know, thankyou!

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (maggiekoesno) [reply portfolio/src/main/webapp/script.js:86]: This doensn't seem to work. 
`Uncaught (in promise) TypeError: commenter.isEmpty is not a function`

**Comment by maggiekoesno** [comment]
I've committed some additional changes. Thanks for the feedback!

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (dhritiwasan14) [reply portfolio/src/main/webapp/script.js]: Ah yes. Thought this was Java for a sec

Comment (dhritiwasan14) [reply portfolio/src/main/webapp/script.js]: My bad! Sorry

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (dhritiwasan14) [reply portfolio/src/main/webapp/script.js]: Just wondering if you could abstract the new comment form section...

**Pull request merged**